lcs/README.md

```diff
@@ -0,0 +1,143 @@
+- **Case-sensitive and case-insensitive** comparison modes
+- **Configurable length limits** to prevent performance issues
+- **Simple existence checking** - check if a similar message already exists
+- **Performance optimized** with O(m*n) time complexity
+
+## Installation
+
+```bash
+go get github.com/rudderlabs/rudder-go-kit/lcs
+```
+
+## Quick Start
+
+```go
+package main
+
+import (
+    "fmt"
+    "github.com/rudderlabs/rudder-go-kit/lcs"
+)
+
+func main() {
+    // Basic similarity calculation
+    similarity := lcs.CalculateSimilarity(
+        "Event name login is not valid must be mapped to one of standard events",
+        "Event name logout is not valid must be mapped to one of standard events",
+    )
+    fmt.Printf("Similarity: %.3f\n", similarity) // Output: Similarity: 0.929
+    
+    // Check if a similar message already exists
+    messages := []string{
+        "Event name login is not valid must be mapped to one of standard events",
+        "Event name logout is not valid must be mapped to one of standard events",
+        "Message type not supported",
+    }
+    
+    newError := "Event name signup is not valid must be mapped to one of standard events"
+    exists := lcs.SimilarMessageExists(newError, messages)
+    fmt.Printf("Similar message exists: %t\n", exists) // Output: true
+}
+```
+
+## API Reference
+
+### Core Functions
+
+#### `CalculateSimilarity(str1, str2 string) float64`
+Calculates similarity between two strings using LCS algorithm.
+Returns a value between 0.0 (no similarity) and 1.0 (identical).
+
+#### `CalculateSimilarityWithOptions(str1, str2 string, opts Options) float64`
+Calculates similarity with custom configuration options.
+
+#### `SimilarMessageExists(target string, messages []string) bool`
+Checks if a similar message already exists in the given set using default options.
+
+#### `SimilarMessageExistsWithOptions(target string, messages []string, opts Options) bool`
+Checks if a similar message exists with custom configuration options.
+
+### Configuration Options
+
+```go
+type Options struct {
+    MaxLength     int     // Maximum character length to process (default: 150)
+    CaseSensitive bool    // Whether to consider case (default: false)
+    Threshold     float64 // Default similarity threshold (default: 0.75)
+    WordBased     bool    // Whether to use word-based comparison (default: true)
+}
+```
+
+
+
+## Algorithm Details
+
+The similarity is calculated using the formula:
+```
+similarity = (2 * LCS_length) / (length1 + length2)
+```
+
+Where:
+- `LCS_length` is the length of the longest common subsequence
+- `length1` and `length2` are the lengths of the input (words or characters)
+
+### Word-based vs Character-based Comparison
+
+**Word-based (default)**: Splits strings into words using `strings.Fields()` and compares word sequences. This is more meaningful for error messages as it focuses on structural similarity rather than character-level differences.
```

Comment:
```suggestion
**Word-based (default)**: Splits strings into words using `strings.Fields()` and compares word sequences. This is more meaningful for performing structural similarity rather than character-level differences.
```

Reply:
addressed